akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -76,6 +76,15 @@ trait Serializer {
   final def fromBinary(bytes: Array[Byte], clazz: Class[_]): AnyRef = fromBinary(bytes, Option(clazz))
 }
 
+object Serializer {
+
+  def manifestFor(s: Serializer, message: AnyRef): Option[String] = s match {
+    case s2: SerializerWithStringManifest ⇒ Some(s2.manifest(message))
+    case _                                ⇒ if (s.includeManifest) Some(message.getClass.getName) else None
+  }
```

Comment:
The original reason why we duplicated this match was to avoid object allocations of `Some`. Nowadays we have allocation free `akka.util.OptionVal` so we can use that here, but then this method should be marked with internal api.

Then it must be used with `match` or `getOrElse` and not things like `map`, `filter`.

Perhaps it would be possible to return `String` instead of `OptionVal` and use `""` for the None case?

Reply:
yeah probably it will work, with String return type, I will try.